projects/data/data/waveforms/rejection.py

```diff
@@ -107,6 +110,19 @@ def rejection_sample(
         num_injections += len(snrs)
         mask = snrs >= snr_threshold
 
+        num_accepted = mask.sum()
+        total_accepted += num_accepted
+        # If we've generated more signals than we need,
+        # figure out where to cut off the arrays
+        if num_signals < total_accepted:
```

Comment:
I'm pretty sure our ledger object support slicing, so I think you could just do 

```
rejected_params = rejected_params[:num_signals]
```
at the end.

If not, I think you can do the slicing after the masking in line 131:

```python
for key, attr in InjectionParameterSet.__dataclass_fields__.items():
    if attr.metadata["kind"] == "parameter":
        rejected[key] = params[key][~mask][:target_accepted]
```

Reply:
Great point, yeah, that's way simpler

Reply:
Wait, no, I don't think this would keep the correct count of rejected samples